# Data overview with pandas

In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('customer_shopping_data.csv')

In [3]:
df.head()

invoice_no customer_id  gender  age  category  quantity    price  \
0    I138884     C241288  Female   28  Clothing         5  1500.40   
1    I317333     C111565    Male   21     Shoes         3  1800.51   
2    I127801     C266599    Male   20  Clothing         1   300.08   
3    I173702     C988172  Female   66     Shoes         5  3000.85   
4    I337046     C189076  Female   53     Books         4    60.60   

  payment_method invoice_date   shopping_mall  
0    Credit Card     5/8/2022          Kanyon  
1     Debit Card   12/12/2021  Forum Istanbul  
2           Cash    9/11/2021       Metrocity  
3    Credit Card   16/05/2021    Metropol AVM  
4           Cash   24/10/2021          Kanyon

In [4]:
df.tail()

invoice_no customer_id  gender  age         category  quantity    price  \
99452    I219422     C441542  Female   45         Souvenir         5    58.65   
99453    I325143     C569580    Male   27  Food & Beverage         2    10.46   
99454    I824010     C103292    Male   63  Food & Beverage         2    10.46   
99455    I702964     C800631    Male   56       Technology         4  4200.00   
99456    I232867     C273973  Female   36         Souvenir         3    35.19   

      payment_method invoice_date     shopping_mall  
99452    Credit Card   21/09/2022            Kanyon  
99453           Cash   22/09/2021    Forum Istanbul  
99454     Debit Card   28/03/2021         Metrocity  
99455           Cash   16/03/2021      Istinye Park  
99456    Credit Card   15/10/2022  Mall of Istanbul

In [5]:
df.describe().T

count        mean         std    min    25%    50%      75%  \
age       99457.0   43.427089   14.990054  18.00  30.00   43.0    56.00   
quantity  99457.0    3.003429    1.413025   1.00   2.00    3.0     4.00   
price     99457.0  689.256321  941.184567   5.23  45.45  203.3  1200.32   

             max  
age         69.0  
quantity     5.0  
price     5250.0

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99457 entries, 0 to 99456
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   invoice_no      99457 non-null  object 
 1   customer_id     99457 non-null  object 
 2   gender          99457 non-null  object 
 3   age             99457 non-null  int64  
 4   category        99457 non-null  object 
 5   quantity        99457 non-null  int64  
 6   price           99457 non-null  float64
 7   payment_method  99457 non-null  object 
 8   invoice_date    99457 non-null  object 
 9   shopping_mall   99457 non-null  object 
dtypes: float64(1), int64(2), object(7)
memory usage: 7.6+ MB


It consists of 3 numeric and 7 categorical variables.

In [7]:
df.isnull().sum()

invoice_no        0
customer_id       0
gender            0
age               0
category          0
quantity          0
price             0
payment_method    0
invoice_date      0
shopping_mall     0
dtype: int64

There are no missing values in our dataset.

In [8]:
df.columns

Index(['invoice_no', 'customer_id', 'gender', 'age', 'category', 'quantity',
       'price', 'payment_method', 'invoice_date', 'shopping_mall'],
      dtype='object')

In [9]:
df.shape

(99457, 10)

Our dataset consists of 99457 observations and 10 attributes.

In [10]:
df.nunique()

invoice_no        99457
customer_id       99457
gender                2
age                  52
category              8
quantity              5
price                40
payment_method        3
invoice_date        797
shopping_mall        10
dtype: int64

In [11]:
df.groupby("shopping_mall").describe().T

shopping_mall   Cevahir AVM  Emaar Square Mall  Forum Istanbul  Istinye Park  \
age      count  4991.000000        4811.000000     4947.000000   9781.000000   
         mean     43.172511          43.561630       43.537497     43.383601   
         std      14.911633          14.889450       15.179075     15.051176   
         min      18.000000          18.000000       18.000000     18.000000   
         25%      30.000000          31.000000       30.000000     30.000000   
         50%      43.000000          43.000000       44.000000     43.000000   
         75%      56.000000          57.000000       57.000000     56.000000   
         max      69.000000          69.000000       69.000000     69.000000   
quantity count  4991.000000        4811.000000     4947.000000   9781.000000   
         mean      2.995191           3.014134        3.002224      3.012473   
         std       1.418168           1.407216        1.419990      1.415532   
         min       1.000000           1.000000        1.000000      1.000000   
         25%       2.000000           2.000000        2.000000      2.000000   
         50%       3.000000           3.000000        3.000000      3.000000   
         75%       4.000000           4.000000        4.000000      4.000000   
         max       5.000000           5.000000        5.000000      5.000000   
price    count  4991.000000        4811.000000     4947.000000   9781.000000   
         mean    687.972719         704.720081      674.363012    686.747525   
         std     952.002169         957.855695      916.138978    932.392502   
         min       5.230000           5.230000        5.230000      5.230000   
         25%      40.660000          45.450000       40.660000     40.660000   
         50%     203.300000         300.080000      203.300000    203.300000   
         75%    1200.320000        1200.320000     1200.320000   1200.320000   
         max    5250.000000        5250.000000     5250.000000   5250.000000   

shopping_mall         Kanyon  Mall of Istanbul     Metrocity  Metropol AVM  \
age      count  19823.000000      19943.000000  15011.000000  10161.000000   
         mean      43.498966         43.440455     43.499301     43.212873   
         std       14.952212         15.038037     14.964396     14.980192   
         min       18.000000         18.000000     18.000000     18.000000   
         25%       31.000000         30.000000     30.000000     30.000000   
         50%       43.000000         43.000000     44.000000     43.000000   
         75%       56.000000         56.500000     56.000000     56.000000   
         max       69.000000         69.000000     69.000000     69.000000   
quantity count  19823.000000      19943.000000  15011.000000  10161.000000   
         mean       2.999395          3.014291      2.990740      3.004626   
         std        1.418310          1.409826      1.406270      1.414937   
         min        1.000000          1.000000      1.000000      1.000000   
         25%        2.000000          2.000000      2.000000      2.000000   
         50%        3.000000          3.000000      3.000000      3.000000   
         75%        4.000000          4.000000      4.000000      4.000000   
         max        5.000000          5.000000      5.000000      5.000000   
price    count  19823.000000      19943.000000  15011.000000  10161.000000   
         mean     691.658944        694.566395    682.831262    682.806120   
         std      951.995108        946.272210    930.235014    923.461864   
         min        5.230000          5.230000      5.230000      5.230000   
         25%       40.660000         45.450000     43.055000     40.660000   
         50%      203.300000        203.300000    203.300000    203.300000   
         75%     1200.320000       1200.320000   1200.320000   1200.320000   
         max     5250.000000       5250.000000   5250.000000   5250.000000   

shopping_mall   Viaport Outlet  Zorlu Center  
a

Most shopping is done in Mall of Istanbul, least shopping is done in Emaar Square Mall. The average age of customers shopping in shopping malls is generally the same (43).

In [12]:
df.groupby("gender").describe().T

gender                Female          Male
age      count  59482.000000  39975.000000
         mean      43.453515     43.387767
         std       14.965468     15.026664
         min       18.000000     18.000000
         25%       30.000000     30.000000
         50%       43.000000     43.000000
         75%       56.000000     56.000000
         max       69.000000     69.000000
quantity count  59482.000000  39975.000000
         mean       3.003581      3.003202
         std        1.411621      1.415129
         min        1.000000      1.000000
         25%        2.000000      2.000000
         50%        3.000000      3.000000
         75%        4.000000      4.000000
         max        5.000000      5.000000
price    count  59482.000000  39975.000000
         mean     688.137615    690.920933
         std      940.791874    941.777893
         min        5.230000      5.230000
         25%       45.450000     40.660000
         50%      203.300000    203.300000
         75%     1200.320000   1200.320000
         max     5250.000000   5250.000000

Based on the gender of the shoppers, the age statistics of male and female shoppers are almost the same, women have done more shopping, and if we look at the average of expenditures, men have spent more.

In [13]:
df.groupby("payment_method").describe().T

payment_method          Cash   Credit Card    Debit Card
age      count  44447.000000  34931.000000  20079.000000
         mean      43.457421     43.427901     43.358534
         std       15.011227     14.997443     14.930696
         min       18.000000     18.000000     18.000000
         25%       30.000000     30.000000     30.000000
         50%       43.000000     43.000000     43.000000
         75%       56.000000     56.000000     56.000000
         max       69.000000     69.000000     69.000000
quantity count  44447.000000  34931.000000  20079.000000
         mean       3.000652      3.007214      3.002988
         std        1.413585      1.410831      1.415654
         min        1.000000      1.000000      1.000000
         25%        2.000000      2.000000      2.000000
         50%        3.000000      3.000000      3.000000
         75%        4.000000      4.000000      4.000000
         max        5.000000      5.000000      5.000000
price    count  44447.000000  34931.000000  20079.000000
         mean     690.823475    688.542467    687.029135
         std      942.599875    943.067511    934.789284
         min        5.230000      5.230000      5.230000
         25%       40.660000     40.660000     45.450000
         50%      203.300000    203.300000    203.300000
         75%     1200.320000   1200.320000   1200.320000
         max     5250.000000   5250.000000   5250.000000

44447 customers completed their purchases with cash, 34931 customers with customer credit cards and 20079 customers with debit cards, if we take the average expenditures as a basis, cash expenditures were relatively higher.

In [14]:
df.groupby("category").describe().T

category              Books      Clothing     Cosmetics  Food & Beverage  \
age      count  4981.000000  34487.000000  15097.000000     14776.000000   
         mean     43.541859     43.409227     43.580645        43.354426   
         std      15.051601     14.998012     14.936556        14.981172   
         min      18.000000     18.000000     18.000000        18.000000   
         25%      30.000000     30.000000     31.000000        31.000000   
         50%      44.000000     43.000000     44.000000        43.000000   
         75%      56.000000     56.000000     56.000000        56.000000   
         max      69.000000     69.000000     69.000000        69.000000   
quantity count  4981.000000  34487.000000  15097.000000     14776.000000   
         mean      3.007830      3.002813      3.011525         2.996548   
         std       1.418658      1.413708      1.413136         1.419154   
         min       1.000000      1.000000      1.000000         1.000000   
         25%       2.000000      2.000000      2.000000         2.000000   
         50%       3.000000      3.000000      3.000000         3.000000   
         75%       4.000000      4.000000      4.000000         4.000000   
         max       5.000000      5.000000      5.000000         5.000000   
price    count  4981.000000  34487.000000  15097.000000     14776.000000   
         mean     45.568621    901.084021    122.448626        15.671948   
         std      21.492662    424.225594     57.458097         7.422176   
         min      15.150000    300.080000     40.660000         5.230000   
         25%      30.300000    600.160000     81.320000        10.460000   
         50%      45.450000    900.240000    121.980000        15.690000   
         75%      60.600000   1200.320000    162.640000        20.920000   
         max      75.750000   1500.400000    203.300000        26.150000   

category               Shoes     Souvenir   Technology          Toys  
age      count  10034.000000  4999.000000  4996.000000  10087.000000  
         mean      43.454953    43.393079    43.368495     43.326262  
         std       15.003153    15.009072    14.835330     15.081875  
         min       18.000000    18.000000    18.000000     18.000000  
         25%       30.000000    30.000000    31.000000     30.000000  
         50%       44.000000    43.000000    43.000000     43.000000  
         75%       57.000000    57.000000    56.000000     56.000000  
         max       69.000000    69.000000    69.000000     69.000000  
quantity count  10034.000000  4999.000000  4996.000000  10087.000000  
         mean       3.011461     2.974795     3.006605      3.005948  
         std        1.408128     1.412148     1.411009      1.405269  
         min        1.000000     1.000000     1.000000      1.000000  
         25%        2.000000     2.000000     2.000000      2.000000  
         50%        3.000000     3.000000     3.000000      3.000000  
         75%        4.000000     4.000000     4.000000      4.000000  
         max        5.000000     5.000000     5.000000      5.000000  
price    count  10034.000000  4999.000000  4996.000000  10087.000000  
         mean    1807.388568    34.894345  3156.935548    107.733185  
         std      845.116269    16.564498  1481.559805     50.364844  
         min      600.170000    11.730000  1050.000000     35.840000  
         25%     1200.340000    23.460000  2100.000000     71.680000  
         50%     1800.510000    35.190000  3150.000000    107.520000  
         75%     2400.680000    46.920000  4200.000000    143.360000  
         max     3000.850000    58.650000  5250.000000    179.200000

Most of the shopping expenses were made for the clothes and the least for the books.